packages/x-components/src/x-modules/tagging/wiring.ts

```diff
@@ -54,6 +70,21 @@ export const setQueryTaggingDebounce = wireCommit('setQueryTaggingDebounce');
  */
 export const setSessionDuration = wireCommit('setSessionDuration');
 
+/**
+ * Tracks the tagging of the query using a debounce which ends if the user
+ * accepts a query.
+ *
+ * @public
+ */
+export const trackWire = moduleDebounce(
+  wireDispatch('track'),
+  ({ state }) => state.config.queryTaggingDebounceMs,
+  [
+    'UserClearedQuery',
+    'UserIsTypingAQuery' // This event is necessary in case there is no instant search
```

Comment:
I wouldn't use this event here. Until there is a newer search, or the search has been cleared somehow, we shouldn't cancel the tracking of the previous one. 

The way we detect user intention is using the search response, not the query in this case.